tests/CMakeLists.txt

```diff
@@ -6,16 +6,128 @@ ADD_DEFINITIONS(-DTESTS_DATADIR="${CMAKE_CURRENT_SOURCE_DIR}/data")
+#
+#   The behaviour of this function depends on :variable:`BUILD_TESTING` option.
+#
+MACRO(ADD_UNIT_TEST NAME SOURCE)
+  CREATE_CTEST_BUILD_TESTS_TARGET()
+
+  IF(NOT BUILD_TESTING)
+    ADD_EXECUTABLE(${NAME} EXCLUDE_FROM_ALL ${SOURCE})
+  ELSE(NOT BUILD_TESTING)
+    ADD_EXECUTABLE(${NAME} ${SOURCE})
+  ENDIF(NOT BUILD_TESTING)
+
+  ADD_DEPENDENCIES(build_tests ${NAME})
+
+  ADD_TEST(${NAME} ${RUNTIME_OUTPUT_DIRECTORY}/${NAME})
+  # Support definition of DYLD_LIBRARY_PATH for OSX systems
+  IF(APPLE)
+    SET_TESTS_PROPERTIES(${NAME} PROPERTIES ENVIRONMENT "DYLD_LIBRARY_PATH=$ENV{DYLD_LIBRARY_PATH}")
+    SET_TESTS_PROPERTIES(${NAME} PROPERTIES ENVIRONMENT "LD_LIBRARY_PATH=$ENV{LD_LIBRARY_PATH}")
+  ENDIF(APPLE)
+
+  IF(NOT BUILD_TESTING)
+    SET_TESTS_PROPERTIES(${NAME} PROPERTIES DEPENDS ctest_build_tests)
+  ENDIF(NOT BUILD_TESTING)
+ENDMACRO(ADD_UNIT_TEST NAME SOURCE)
+
+#.rst:
+# .. command:: ADD_PYTHON_UNIT_TEST (NAME SOURCE [MODULES...])
+#
+#   Add a test called `NAME` that runs an equivalent of ``python ${SOURCE}``,
+#   optionnaly with a `PYTHONPATH` set to `CMAKE_BINARY_DIR/MODULE_PATH` for each MODULES
+#   `SOURCE` is relative to `PROJECT_SOURCE_DIR`
+#
+#   .. note:: :command:`FINDPYTHON` should have been called first.
+#
+MACRO(ADD_PYTHON_UNIT_TEST NAME SOURCE)
+  ADD_TEST(NAME ${NAME} COMMAND ${PYTHON_EXECUTABLE} "${PROJECT_SOURCE_DIR}/${SOURCE}")
+  SET(PYTHONPATH)
+
+  SET(MODULES "${ARGN}")  # ARGN is not a variable
+  FOREACH(MODULE_PATH IN LISTS MODULES)
+    LIST(APPEND PYTHONPATH "${CMAKE_BINARY_DIR}/${MODULE_PATH}")
+  ENDFOREACH(MODULE_PATH IN LISTS MODULES)
+
+  IF(DEFINED ENV{PYTHONPATH})
+    LIST(APPEND PYTHONPATH "$ENV{PYTHONPATH}")
+  ENDIF(DEFINED ENV{PYTHONPATH})
+
+  # get path separator to join those paths
+  EXECUTE_PROCESS(COMMAND
+      "${PYTHON_EXECUTABLE}" "-c" "import os; print(os.pathsep)"
+      OUTPUT_VARIABLE PATHSEP
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  STRING(REPLACE ";" "${PATHSEP}" PYTHONPATH_STR "${PYTHONPATH}")
+  SET(ENV_VARIABLES "PYTHONPATH=${PYTHONPATH_STR}")
+  IF(APPLE)
+    LIST(APPEND ENV_VARIABLES "LD_LIBRARY_PATH=$ENV{LD_LIBRARY_PATH}")
+    LIST(APPEND ENV_VARIABLES "DYLD_LIBRARY_PATH=$ENV{DYLD_LIBRARY_PATH}")
+  ENDIF(APPLE)
+  SET_TESTS_PROPERTIES(${NAME} PROPERTIES ENVIRONMENT "${ENV_VARIABLES}")
+ENDMACRO(ADD_PYTHON_UNIT_TEST NAME SOURCE)
+
+# DEFINE_UNIT_TEST(NAME LIB)
+# ----------------------
+#
+# Compile a program and add it as a test
+#
+MACRO(DEFINE_UNIT_TEST NAME LIB)
+  ADD_UNIT_TEST(${NAME} ${NAME}.cc)
+  TARGET_INCLUDE_DIRECTORIES(${NAME} SYSTEM PUBLIC ${EIGEN3_INCLUDE_DIR})
+  TARGET_LINK_LIBRARIES(${NAME} ${PUBLIC_KEYWORD} ${LIB})
+ENDMACRO(DEFINE_UNIT_TEST)
+
 MACRO(DYNAMIC_GRAPH_TEST NAME)
   ADD_UNIT_TEST(${NAME} ${NAME}.cpp)
+  TARGET_INCLUDE_DIRECTORIES(${NAME} SYSTEM PUBLIC ${EIGEN3_INCLUDE_DIR})
   TARGET_LINK_LIBRARIES(${NAME} ${PROJECT_NAME} ${Boost_LIBRARIES})
 ENDMACRO(DYNAMIC_GRAPH_TEST)
 
+SET(CMAKE_VERBOSE_MAKEFILE True)
```

Comment:
If you need this variable to be defined, please configure it on your command line instead of in the file, eg.: `cmake -DCMAKE_VERBOSE_MAKEFILE=True ..`